### ISSUE TITLE
Indent rule: "first" option for VariableDeclarator doesn't work with tabs

### ISSUE THREAD

**Issue by scottohara** [eslint/eslint#11213]
**Tell us about your environment**

* **ESLint Version: 5.11.1**
* **Node Version: 11.5.0**
* **npm Version: 6.5.0**

**What parser (default, Babel-ESLint, etc.) are you using?**
typescript-eslint-parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  ...
  "parser": "typescript-eslint-parser",
  "plugins": [
    "typescript"
  ],
  "rules": {
    "indent": ["error", "tab", {
      "VariableDeclarator": {
        "var": "first",
        "let": "first",
        "const": "first"
      }
    }
    ...
  }
}; 
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
  private keyHandler(event: KeyboardEvent): void {
    const INSERT_KEY = 45,
          N_KEY = 78;

    ...
  }
```

```bash
eslint '*.ts'
```

**What did you expect to happen?**
No errors.

The docs state that the value `first` (added in v5.11.0) means "all the declarators should be aligned with the first declarator".

When using tabs instead of spaces, the indent count does not seem to be correct.

Below is the above snippet of code, marked up with `^` to represent tab characters (and column numbers at the top):

(assume a tab width of 2)

```js
12345678901  <- column
^ private keyHandler(event: KeyboardEvent): void {
^ ^ const^INSERT_KEY = 45,
^ ^ ^ ^ ^ N_KEY = 78;
```

The line with the `const` declaration begins with 2 tabs, followed by the five characters of the word `const`, then another tab.  This means that the `INSERT_KEY` symbol starts at column 11, preceded by 8 characters.

The `N_KEY` symbol is preceded by 5 tabs, so it too starts at column 11.

I would argue that this matches the docs description of being aligned with the first declarator, yet the rule returns an error as shown below.

**What actually happened? Please include the actual, raw output from ESLint.**
Expected indentation of 8 tabs but found 5.

**Are you willing to submit a pull request to fix this bug?**
If it is in fact a bug and not intentional, sure.

**Comment by not-an-aardvark** [comment]
Hi, thanks for the report. This is working as intended, although it's admittedly not ideal -- the `first` options in the indent rule are generally intended for when you're using spaces to indent your code rather than tabs. (You're arguing that the line should be indented with 5 tabs in order to be aligned, but that would only make sense if the user happens to be using a tab width of 2 spaces, and the `indent` rule doesn't know anything about what tab width you're using in your editor.)

Eventually, I think it would be nice if the `indent` rule supports "alignment" spaces for the `first` case (e.g. ensuring that the line in your example is indented with 2 tabs followed by 6 spaces), which would ensure that the appropriate characters are always aligned regardless of your tab width. This doesn't seem like it would be too difficult to implement, although I haven't seen much support for a proposal like that, presumably because not very many people use tabs to indent their code.
